src/cryptography/hazmat/bindings/openssl/binding.py

```diff
@@ -81,7 +77,20 @@ def _register_osrandom_engine(cls):
             result = cls.lib.ENGINE_set_RAND(engine, cls._osrandom_method)
             assert result == 1
             result = cls.lib.ENGINE_add(engine)
-            assert result == 1
+            if result != 1:
+                # Engine already added. Clear the error stack.
+                errors = []
+                while True:
+                    code = cls.lib.ERR_get_error()
+                    if code == 0:
+                        break
+
+                    errors.append(code)
```

Comment:
This code already exists in the backend, can we factor it into a standalone function that is used both here and there?